compiler/src/iree/compiler/Codegen/Common/MaterializeTuningSpecsPass.cpp

```diff
@@ -138,8 +148,25 @@ getDefaultTuningSpec(ModuleOp module,
 
   // Load the library through the codegen dialect so that we cache the parsed
   // module.
-  return dialect.getOrParseTransformLibraryModule(defaultTuningSpecName,
-                                                  *defaultTuningSpecSource);
+  FailureOr<ModuleOp> defaultTransformLibrary =
+      dialect.getOrParseTransformLibraryModule(defaultTuningSpecName,
+                                               *defaultTuningSpecSource);
+  if (failed(defaultTransformLibrary)) {
+    return module->emitError()
+           << "Failed to parse default tuning spec transform library for '"
+           << arch << "'";
+  }
+  // Iterate through operations and validate their attributes
+  for (Operation &op : (*defaultTransformLibrary).getBody()->getOperations()) {
+    for (NamedAttribute attr : op.getAttrs()) {
+      if (failed(dialect.verifyOperationAttribute(&op, attr))) {
+        return op.emitError() << "Attribute verification failed for operation "
+                                 "in default tuning spec";
+      }
+    }
+  }
```

Comment:
I don't think we should verify default specs -- these are already verified by our tests when building the compiler. We could do that but under debug builds only.